graphql_compiler/tests/test_compiler.py

```diff
@@ -660,6 +660,45 @@ def test_filter_in_optional_block(self):
 
         check_test_data(self, test_data, expected_match, expected_gremlin, expected_sql)
 
+    def test_filter_in_optional_and_count(self):
+        test_data = test_input_data.filter_in_optional_and_count()
+
+        expected_match = '''
+            SELECT
+                Species___1.name AS `species_name`
+            FROM (
+                MATCH {{
+                    class: Species,
+                    as: Species___1
+                }}.in('Animal_OfSpecies') {{
+                   where: ((name = {animal_name})),
+                   optional: true,
+                   as: Species__in_Animal_OfSpecies___1
+                }}
+                RETURN $matches
+            )
+            LET $Species___1___in_Species_Eats = Species___1.in("Species_Eats").asList()
+                WHERE (
```

Comment:
De-indent the `WHERE` and the expression inside it, it is at the same level as the `LET` and `FROM` clauses.